examples/pix2pose/pose_metrics_test.py

```diff
@@ -0,0 +1,159 @@
+import pytest
+import cv2
+import os
+from tensorflow.keras.utils import get_file
+from paz.backend.image import load_image
+from paz.backend.camera import Camera
+from paz.pipelines import PIX2POSEPowerDrill
+from paz.backend.groups.quaternion import quaternion_to_rotation_matrix, \
+    rotation_matrix_to_quaternion
+import numpy as np
+from paz.backend.groups import build_rotation_matrix_x
+from paz.abstract.messages import Pose6D
+from examples.pix2pose.eval import compute_ADI, compute_ADD, \
+    transform_points3D, get_translation, get_quaternion
+
+
+@pytest.fixture
+def image_with_powerdrill():
+    URL = ('https://github.com/oarriaga/altamira-data/releases/download'
+           '/v0.9.1/image_with_YCB_objects.jpg')
+    filename = os.path.basename(URL)
+    fullpath = get_file(filename, URL, cache_subdir='paz/tests')
+    image = load_image(fullpath)
+    return image
+
+
+@pytest.fixture
+def pose_A():
+    rotation = np.array(
+        [-0.9078519116941933, -0.4175345487208392, 0.038330959211728285,
+         0.3050982069800776, -0.5951250307633585,
+         0.743466715795677, -0.28761117420744436, 0.6866522290644125,
+         0.6676740178175343]).reshape(3, 3)
+    translation = np.array([104.54312464515644, 71.49346046440272, 955.3250389603148]).reshape(3,1)                                                                            
+    return rotation, translation
+
+
+@pytest.fixture    
+def pose_B():
+    rotation = np.array(
+        [-0.9078519116941933, -0.4175345487208392, 0.038330959211728285,
+         0.3050982069800776, -0.5951250307633585,
+         0.743466715795677, -0.28761117420744436, 0.6866522290644125,
+         0.6676740178175343]).reshape(3, 3)
+    translation = np.array([114.54312464515644, 71.49346046440272, 955.3250389603148]).reshape(3,1)                                                                       
+    return rotation, translation
+
+
+@pytest.fixture
+def pose_C(pose_A):
+    rotation, translation= pose_A[0], pose_A[1]
+    rotation_x=build_rotation_matrix_x(20)
+    transformed_rotation=rotation_x @ rotation                                                                      
+    return transformed_rotation, translation
+
+
+@pytest.fixture
+def pix2pose_pose6d(image_with_powerdrill):
+    cv2.ocl.setUseOpenCL(False)
+    cv2.setNumThreads(1)
+    cv2.setRNGSeed(777)
+    camera = Camera()
+    camera.intrinsics_from_HFOV(55, (480, 640, 3))
+    detector = PIX2POSEPowerDrill(camera, score_thresh=0.45, offsets=[0.25, 0.25], epsilon=0.015)
+    inferences = detector(image_with_powerdrill)
+    pose6D = inferences['poses6D'][0]
+    return pose6D
+
+
+@pytest.fixture
+def points3D():
+    X=np.linspace(1e-1,1e-3,  num=9128).reshape(9128,1)
+    Y=np.linspace(1e-2,1e-4,  num=9128).reshape(9128,1)
+    Z=np.linspace(1e-4,1e-6,  num=9128).reshape(9128,1)
```

Comment:
aren't these values to small? I don't think they correspond to the usual expected values. Maybe you can make another test that uses values closer to the real ones.

Reply:
resolved